ci/kind/kind-setup.sh

```diff
@@ -348,73 +398,127 @@ while [[ $# -gt 0 ]]
 
   case $key in
     create)
-      CLUSTER_NAME="$2"
-      shift 2
+      ACTION="create"
+      shift
       ;;
     destroy)
-      CLUSTER_NAME="$2"
-      destroy
-      exit 0
+      ACTION="destroy"
+      shift
       ;;
     --pod-cidr)
+      add_option "--pod-cidr" "create"
       POD_CIDR="$2"
       shift 2
       ;;
     --service-cidr)
+      add_option "--service-cidr" "create"
       SERVICE_CIDR="$2"
       shift 2
       ;;
     --ip-family)
+      add_option "--ip-family" "create"
       IP_FAMILY="$2"
       shift 2
       ;;
     --encap-mode)
+      add_option "--encap-mode" "create"
       ENCAP_MODE="$2"
       shift 2
       ;;
     --no-proxy)
+      add_option "--no-proxy" "create"
       PROXY=false
       shift
       ;;
     --no-kube-proxy)
+      add_option "--no-kube-proxy" "create"
       KUBE_PROXY_MODE="none"
       shift
       ;;
     --prometheus)
+      add_option "--prometheus" "create"
       PROMETHEUS=true
       shift
       ;;
     --subnets)
+      add_option "--subnets" "create"
       SUBNETS="$2"
       shift 2
       ;;
     --images)
+      add_option "--image" "create"
       IMAGES="$2"
       shift 2
       ;;
     --antrea-cni)
+      add_option "--antrea-cni" "create"
       ANTREA_CNI=true
       shift
       ;;
     --num-workers)
+     add_option "--num-workers" "create"
       NUM_WORKERS="$2"
       shift 2
       ;;
     --k8s-version)
+      add_option "--k8s-version" "create"
       K8S_VERSION="$2"
       shift 2
       ;;
+    --all)
+      add_option "--all" "destroy"
+      CLUSTER_NAME="*"
+      shift
+      ;;
+    --until)
+      add_option "--until" "destroy"
+      UNTIL_TIME_IN_MINS="$2"
+      shift 2
+      ;;
     help)
       print_usage
       exit 0
       ;;
-    *)    # unknown option
+    -*)    # unknown option
       echoerr "Unknown option $1"
       exit 1
       ;;
+    *)    # positional arg
+      positional_args+=("$1")
+      shift
+      ;;
  esac
  done
 
+ check_action
+
+if (( ${#positional_args[@]} > 1 )); then
+    echoerr "Too many positional arguments, only expected one (cluster name)"
+    exit 1
+fi
+
+if (( ${#positional_args[@]} == 1 )) && [[ "$CLUSTER_NAME" == "*" ]]; then
+    echoerr "Cannot specify cluster name when using --all"
+    exit 1
+fi
+
+if (( ${#positional_args[@]} == 1 )); then
+    CLUSTER_NAME=${positional_args[0]}
+fi
+
+if [[ -z "$CLUSTER_NAME" ]]; then
+    echoerr "Missing cluster name"
+    exit 1
+fi
+
+if [[ $ACTION == "create" ]]; then
+      create
+fi
+
+if [[ $ACTION == "destroy" ]]; then
+      destroy
+fi
+
 kind_version=$(kind version | awk  '{print $2}')
```

Comment:
it feels to me that this code (Kind version check) should be earlier in the script
The check needs to be performed earlier, before `create` is called